consensus/src/block_storage/block_store.rs

```diff
@@ -482,11 +482,16 @@ impl BlockStore {
     }
 
     pub fn pipeline_pending_latency(&self, proposal_timestamp: Duration) -> Duration {
-        let ordered_round = self.ordered_root().round();
-        let commit_round = self.commit_root().round();
-        let pending_rounds = ordered_round.checked_sub(commit_round).unwrap();
-        let ordered_timestamp = Duration::from_micros(self.ordered_root().timestamp_usecs());
-        let committed_timestamp = Duration::from_micros(self.commit_root().timestamp_usecs());
+        let ordered_root = self.ordered_root();
+        let commit_root = self.commit_root();
+        let pending_path = self.path_from_commit_root(self.ordered_root().id()).unwrap_or_default();
+
+        let ordered_round = ordered_root.round();
+        let commit_round = commit_root.round();
+        let pending_rounds = pending_path.len();
+        let ordered_timestamp = Duration::from_micros(ordered_root.timestamp_usecs());
+        let oldest_not_committed_timestamp = pending_path.last().map(|b| Duration::from_micros(b.timestamp_usecs())).unwrap_or(Duration::ZERO);
```

Comment:
are we sure about the order here? I see the code here https://github.com/aptos-labs/aptos-core/blob/88c987c2d14a8fbfffac004192b8304b2871024d/consensus/src/block_storage/block_tree.rs#L394 doesn't match the comments https://github.com/aptos-labs/aptos-core/blob/7a5903a52dffd5595c34b29c671a571a46dd1408/consensus/src/block_storage/mod.rs#L36

Reply:
should I depend on order, or just take min?

Reply:
min seems a safer choice 😄 iterating through this should be negligible